lib/Exporter/Writer/KnpSnappyWriter.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+/*
+ * This file is part of the Sonata package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Exporter\Writer;
+
+use Knp\Snappy\Pdf;
+
+/**
+ * Export data in Pdf format.
+ *
+ * Full list of available options
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class KnpSnappyWriter implements WriterInterface
+{
+    /**
+     * @var string
+     */
+    protected $filename;
+
+    /**
+     * @var string
+     */
+    protected $wktmltopdfBinary;
+
+    /**
+     * @var bool
+     */
+    protected $showHeaders;
+
+    /**
+     * @var bool
+     */
+    protected $showBorders;
+
+    /**
+     * @var array
+     */
+    protected $snappyOptions;
+
+    /**
+     * @var int
+     */
+    protected $position;
+
+    /**
+     * @var @resource
```

Comment:
@OskarStark, I totally refactored it since this code is ugly.
I'll submit a new one.